### DISCUSSION TITLE
Waypoint formula construction question

### DISCUSSION THREAD

**Discussion by Lineflyer** [cgeo/cgeo#13659] [Q&A] [answered]
Today I totally failed to use the variables together with a calculated wapoint...need your help to find my failure:

Cache: https://coord.info/GC70B35

Final formula from cache description:
N 51° 20.000+A+C+E+G+I E 07° 06.000+B+D+F+H+J

The "problem" is, that the sum of variable in each coordinates seconds lead to an overflow into the minute.

For the seconds I used the modulo operator, which works...but how to get the "overflow digit" into the minute?
What I have so far....
N51° 2((A+C+E+G+I)/1000).((A+C+E+G+I)%1000)__'    E 007° 0((B+D+F+H+J)/1000).((B+D+F+H+J)%1000)__'

It works for seconds but of course not for the minute as ((A+C+E+G+I)/1000) returns some decimal digits. I need the reverse operator of modulo...

**Comment by eddiemuc** [comment, marked as answer]
If I understand the problem correctly, the following should work:

N 51° (20000+A+C+E+G+I)/1000

E 07° (6000+B+D+F+H+J)/1000


You should not try to handle the parts before and after the minute decimal as two separate things. It's just one decimal number, not two integer numbers (and the part after the dot are NOT seconds, they are minute fractions)

For sake of completeness: there's the function `round` which you can use to round a decimal number. There's no function `trunc` yet, but it can be added. However, both are not necessary imho to solve your problem.

**Comment by eddiemuc** [comment]
(I edited my previous comment)

**Comment by Lineflyer** [comment]
@eddiemuc 
Thanks...have to test that. AFAIR I tried something similar, but it started to use DD.DDDD format then which does not provide the desired DD.MMM result of course.

You are right of course with regard to "Minute fractions".

**Comment by eddiemuc** [comment]
If this changes format to DD.DDD I wou5be interested in the exact input. There might be a bug hiding in there...

I had a second idea which might be more straightforward:

N 51° 20.000+(A+C+E+G+I)/1000

E 07° 06.000+(B+D+F+H+J)/1000

This might be closer to what the owner meant (but failed to express in a mathematically sound way)

**Comment by Lineflyer** [comment]
> If this changes format to DD.DDD I wou5be interested in the exact input. There might be a bug hiding in there...

I started in the guided mode...might be the reason.

Anyway, will need to test it again next days to see how it works or whether there is any problem.

**Comment by Lineflyer** [comment]
> N 51° 20.000+(A+C+E+G+I)/1000
>
> E 07° 06.000+(B+D+F+H+J)/1000

That results in (example):
N 51° 20.1.123 instead of N51° 21.123
Same for longitude

> N 51° (20000+A+C+E+G+I)/1000
>
> E 07° (6000+B+D+F+H+J)/1000

That however works as desired.
After seeing it, this indeed seems the logical and mathematical solution. I was indeed thinking to complicated.

**Reply by eddiemuc** [reply]
> N 51° 20.000+(A+C+E+G+I)/1000
> That results in (example):
> N 51° 20.1.123 instead of N51° 21.123

Phew this will get complex.

Before I tackle this: is it wanted that formula results for minutes should spill over after 3 digits?

Also e.g. here?

N51 AA.B+C , A=1,B=200, C=900
--> should result in N51 12.100 ? Or in N51 11.110?

What about DD.DDDDD? Should formula results for digit part spill over after 5 digits there as well?

**Reply by ztNFny** [reply]
> > N 51° 20.000+(A+C+E+G+I)/1000
> > That results in (example):
> > N 51° 20.1.123 instead of N51° 21.123
> 
> Phew this will get complex.
> 
> Before I tackle this: is it wanted that formula results for minutes should spill over after 3 digits?

The current result looks perfectly fine to me.

Most caches don't just spill over, so in most cases such a result is an indicator that something is wrong. Silently handling this would take the info away from the user.

**Comment by Lineflyer** [comment]
> Phew this will get complex.

I also think the result is acceptable and the other formula was already a solution for my use case. So IMHO no need for a change unless you see an error there.

**Reply by eddiemuc** [reply]
No error in the implementation in the sense that it works as I intended to program it (although it took me a while to understand that myself :-)). So then let's close here
